File: addons/mail/static/src/assets/mail-template-component/fields/mail-template.js
/** @odoo-module **/

import { Define } from '@mail/define';

export default Define`
    {Record/insert}
        [Record/models]
            Field
        [Field/name]
            mailTemplate
        [Field/model]
            MailTemplateComponent
        [Field/type]
            one
        [Field/target]
            MailTemplate
        [Field/isRequired]
            true
`;

File: addons/pos_self_order/models/mail_template.py
from odoo import _, api, models

class MailTemplate(models.Model):
    _name = 'mail.template'
    _inherit = ['mail.template', 'pos.load.mixin']

    @api.model
    def _load_pos_data_domain(self, data, config):
        return [('id', 'in', [preset['mail_template_id'] for preset in data['pos.preset']])]

    @api.model
    def _load_pos_data_fields(self, config):
        return ['id']
